client/app/services/style.service.js

```diff
@@ -0,0 +1,84 @@
+(function () {
+    'use strict';
+    /**
+     * @fileoverview This file provides a OpenLayers style service.
+     * It returns a styleFunction depending on the feature's attributes
+     * StyleFunction docs can be found here: http://openlayers.org/en/latest/apidoc/ol.style.html#.StyleFunction
+     *
+     * The following properties are used for styling
+     *  - type
+     *  - selected
+     *  - fill (in rgba)
+     *  - stroke (in rgba)
+     *  - strokeWidth
+     *  - strokeStyle (dashed / dotted / none)
+     *  - showLineArrows (true / false)
+     *  - fontSize
+     *  - fontWeight
+     *  - labelRotation (in degrees)
+     *  - labelText
+     *  - pointSize
+     *  - iconName
+     *  - showMeasurement (true / false)
+     *  - measureText
+     */
+
+    angular
+        .module('taskingManager')
+        .service('styleService', [styleService]);
+
+    function styleService() {
+
+        var service = {
+            getTaskStyleFunction: getTaskStyleFunction,
+        };
+
+        return service;
+
+        /**
+         * Creates a StyleFunction for tasks based on the feature's taskStatus and taskLocked properties
+         * @param feature
+         * @returns {Array}
+         */
+        function getTaskStyleFunction(feature){
+
+            // Get the feature's properties that control styling
+            var status = feature.get('taskStatus');
+            var isLocked = feature.get('taskLocked');
+
+            // calculate the fill colour and opacity settings based on status, use rgba because this is the way to
+            // set opacity in OL3, but also better for cross browser than named colors
+            var fillColor = null;
+            if (status === 'READY') {
+                fillColor  = 'rgba(223,223,223,0.1)';//very light gray, 0.1 opacity
```

Comment:
make these fill colours variables (e.g. fillColourReady = 'rgba(.....)' at the top so it is easier to find and change them